icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -923,6 +1034,36 @@ class Parser {
                 singleUnitResult.index = token.getSimpleUnitIndex();
                 break;
 
+            case Token::TYPE_ALIAS: {
+                auto aliasIndex = token.getAliasIndex();
+                if (aliasIndex < 0 || aliasIndex >= gUnitAliases.length()) {
+                    status = kUnitIdentifierSyntaxError;
+                    return {};
+                }
+                auto replacement = gUnitReplacements[aliasIndex];
+
+                // Create new source string: replacement + remaining unparsed portion
+                fModifiedSource.clear();
```

Comment:
If I understand correctly, you've incorporated the alias mechanism directly into your parser, but then, if you encounter a token with an alias, you just create a new version of the unit string and copy the alias's replacement text into the string you're parsing and continue parsing the _new_ string instead.  I wonder if it'd make things simpler to handle alias replacement in a separate preprocessing step instead of incorporating it directly into your parser?

(I'm not completely sold on this suggestion-- it may actually be simpler the way you're doing it now.)

Reply:
Yeah, I think you're right.  Thanks for enumerating all this.